cluster-autoscaler/proposals/pricing.md

```diff
@@ -0,0 +1,177 @@
+# Cost-based node group ranking function for Cluster Autoscaler
+##### Author: mwielgus
+ 
+# Introduction
+ 
+Cluster autoscaler tries to increase the cluster size if there are some pods that are unable to fit into the nodes currently present in the cluster. If there are two or more types of nodes in the cluster CA has to decide which one to add. At this moment it, by default, just picks a random one (there are some other expanders but they are also relatively simple).
```

Comment:
Maybe I picked the wrong words. Everything is better than random, but least-waste doesn't take into account many aspects that this approach is trying to address. Like preemptive vs non-preemptive, instances with gpu/ssd vs instances without, etc. So I called it relatively simple.